include/nbl/asset/utils/ICPUVirtualTexture.h

```diff
@@ -528,9 +528,38 @@ class ICPUVirtualTexture final : public IVirtualTexture<ICPUImageView, ICPUSampl
         return getDSlayoutBindings_internal<ICPUDescriptorSetLayout>(_outBindings, _outSamplers, _pgtBinding, _fsamplersBinding, _isamplersBinding, _usamplersBinding);
     }
 
-    auto getDescriptorSetWrites(ICPUDescriptorSet::SWriteDescriptorSet* _outWrites, ICPUDescriptorSet::SDescriptorInfo* _outInfo, ICPUDescriptorSet* _dstSet, uint32_t _pgtBinding = 0u, uint32_t _fsamplersBinding = 1u, uint32_t _isamplersBinding = 2u, uint32_t _usamplersBinding = 3u) const
+    bool updateDescriptorSet(ICPUDescriptorSet* _dstSet, uint32_t _pgtBinding = 0u, uint32_t _fsamplersBinding = 1u, uint32_t _isamplersBinding = 2u, uint32_t _usamplersBinding = 3u) const
     {
-        return getDescriptorSetWrites_internal<ICPUDescriptorSet>(_outWrites, _outInfo, _dstSet, _pgtBinding, _fsamplersBinding, _isamplersBinding, _usamplersBinding);
+        // Update _pgtBinding.
+        {
+            auto pgtInfos = _dstSet->getDescriptorInfos(_pgtBinding, IDescriptor::E_TYPE::ET_COMBINED_IMAGE_SAMPLER);
+            if (pgtInfos.empty())
+                return false;
+
+            assert(pgtInfos.size() == 1ull);
+            auto& info = pgtInfos.begin()[0];
+            info.info.image.imageLayout = IImage::EL_UNDEFINED;
+            info.info.image.sampler = nullptr;
+            info.desc = core::smart_refctd_ptr<ICPUImageView>(getPageTableView());
+        }
+
+        auto updateSamplersBinding = [&](const uint32_t binding, const auto& views)
+        {
+            auto infos = _dstSet->getDescriptorInfos(binding, IDescriptor::E_TYPE::ET_COMBINED_IMAGE_SAMPLER);
+
+            for (uint32_t i = 0; i < infos.size(); ++i)
```

Comment:
I'd validate that `infos.size()>=views.size()` and log + return error if not

Reply:
Again, `return`-ed `false`, but we don't have a logger here.

Reply:
Resolved?

Reply:
yes